js/shared.js

```diff
@@ -372,6 +372,82 @@ function undoBrowserSelection(s){
     s?.removeAllRanges?.() ?? s?.empty?.()
 }
 
+/** 
+ * Discover proper identifier for manuscript or suggest new IRI. 
+ *
+ * @async
+ * @function findShelfmark
+ * @param { string } msid Identifier to match or validate.
+ * @param { boolean } [forceNew] True to skip search for collision. *risky*
+ * @returns Valid msid string or parent IRI
+ */ 
+async function findShelfmark(msid, forceNew) {
+    try {
+        // wash msid
+        if (typeof msid !== 'string') {
+            const invalidInputEvent = new CustomEvent("Failed to Query Rerum. Invalid msid identifier input.")
+            globalFeedbackBlip(invalidInputEvent, 'Failed to query for Shelfmark: Attempted to add a non string.', false)
+            return
+        }
+
+        const cleanMsid = msid.replace(/[@$%*?]+/g, '') 
+        .replace(/\s+/g, ' ') 
+        .trim() 
+
+        if (cleanMsid.length === 0) {
+            const invalidInputEvent = new CustomEvent("Failed to Query Rerum. Invalid msid identifier input.")
+            globalFeedbackBlip(invalidInputEvent, 'Failed to query for Shelfmark: Attempted to add an empty string.', false)
+            return
+        }
+
+        // return washed msid or parent msid
+        if (forceNew) {
+            return cleanMsid
+        }
+        else {
+            // check for parent msid
+            const query = {
+                "body": { 
```

Comment:
mongo queries cannot have embedded objects like this. You need it to look like `__rerum.generatedBy` to find a result.